src/parse.rs

```diff
@@ -855,13 +856,13 @@ impl<'a> FirstPass<'a> {
                     begin_text = ix + 1;
                     LoopInstruction::ContinueAndSkip(0)
                 }
-                b'&' => match scan_entity(&bytes[ix..]) {
+                b'&' => match scanner.scan_entity(&bytes[ix..]) {
                     (n, Some(value)) => {
                         self.tree.append_text(begin_text, ix);
                         self.tree.append(Item {
                             start: ix,
                             end: ix + n,
-                            body: ItemBody::SynthesizeText(self.allocs.allocate_cow(value)),
+                            body: ItemBody::SynthesizeText(self.allocs.allocate_cow(value.to_string().into())),
```

Comment:
Codepoint `&str`s are transient so this unfortunately requires an allocation. I don't know what the `self.allocs` is for at this point, but I think this might be avoidable.